examples/geometry/src/components/polyline.tsx

```diff
@@ -0,0 +1,131 @@
+/* eslint-disable complexity */
+import {
+  forwardRef,
+  useContext,
+  useEffect,
+  useImperativeHandle,
+  useMemo,
+  useRef
+} from 'react';
+
+import {GoogleMapsContext, useMapsLibrary} from '@vis.gl/react-google-maps';
+
+import type {Ref} from 'react';
+
+type PolylineEventProps = {
+  onClick?: (e: google.maps.MapMouseEvent) => void;
+  onDrag?: (e: google.maps.MapMouseEvent) => void;
+  onDragStart?: (e: google.maps.MapMouseEvent) => void;
+  onDragEnd?: (e: google.maps.MapMouseEvent) => void;
+  onMouseOver?: (e: google.maps.MapMouseEvent) => void;
+  onMouseOut?: (e: google.maps.MapMouseEvent) => void;
+};
+
+type PolylineCustomProps = {
+  /**
+   * this is an encoded string for the path, will be decoded and used as a path
+   */
+  encodedPath?: string;
+};
+
+export type PolylineProps = google.maps.PolylineOptions &
+  PolylineEventProps &
+  PolylineCustomProps;
+
+export type PolylineRef = Ref<google.maps.Polyline | null>;
+
+function usePolyline(props: PolylineProps) {
+  const {
+    onClick,
+    onDrag,
+    onDragStart,
+    onDragEnd,
+    onMouseOver,
+    onMouseOut,
+    encodedPath,
+    ...polylineOptions
+  } = props;
+  // This is here to avoid triggering the useEffect below when the callbacks change (which happen if the user didn't memoize them)
```

Comment:
I like this technique, never thought about it this way. 
It's technically iccorrect, but works better than the default for 95% of use-cases.

Reply:
I just copied this from the Polygon example 😅